crates/apollo-compiler/src/parser.rs

```diff
@@ -257,8 +284,16 @@ impl SourceFile {
     }
 
     pub fn ariadne(&self) -> &ariadne::Source {
-        self.ariadne
-            .get_or_init(|| ariadne::Source::from(&self.source_text))
+        &self
+            .source
+            .get_or_init(|| MappedSource::new(&self.source_text))
+            .ariadne
+    }
+
+    pub fn get_line_column(&self, index: usize) -> Option<(usize, usize)> {
+        let char_index = self.source.get()?.map_index(index);
```

Comment:
`OnceLock::get` returns `None` if the value doesn’t happen to already be initialized. We want to call `get_or_init` instead, or something that does (see method above)